x-pack/plugin/sql/src/test/java/org/elasticsearch/xpack/sql/planner/QueryTranslatorTests.java

```diff
@@ -2074,15 +2074,15 @@ public void testNoCountDoesNotTrackHits() {
 
     public void testZonedDateTimeInScripts() {
         PhysicalPlan p = optimizeAndPlan(
-                "SELECT date FROM test WHERE date + INTERVAL 1 YEAR > CAST('2019-03-11T12:34:56.000Z' AS DATETIME)");
+                "SELECT date FROM test WHERE date - INTERVAL 999999999 YEAR > CAST('2019-03-11T12:34:56.000Z' AS DATETIME)");
         assertEquals(EsQueryExec.class, p.getClass());
         EsQueryExec eqe = (EsQueryExec) p;
         assertThat(eqe.queryContainer().toString().replaceAll("\\s+", ""), containsString(
                 "\"script\":{\"script\":{\"source\":\"InternalQlScriptUtils.nullSafeFilter("
-                        + "InternalQlScriptUtils.gt(InternalSqlScriptUtils.add(InternalQlScriptUtils.docValue(doc,params.v0),"
+                        + "InternalQlScriptUtils.gt(InternalSqlScriptUtils.sub(InternalQlScriptUtils.docValue(doc,params.v0),"
```

Comment:
I've changed the test from an addition to a subtraction, as well as used a max value, so that the optimisation is prevented. Consequently the translation had to be adjusted.

Reply:
Could you then add a test where the optimization is applied? I'm confused since in this instance, I expect the PR to add tests not modify existing ones.

Reply:
:-) Sure, tests have been added.
This test however still needs to be changed (as is now), as it meant to test `testZonedDateTimeInScripts`, not optimisations. What it did was the integer similar of `x + 2  > 3` which the new optimisation rule would translate to `x > 1`; so it would simplify the test, reducing its testing value. My change only made sure the literals non-field values are such that the optimisation won't take place, so that the test retains its original intention (sort of `x - MAX > 3`, just like [here](https://github.com/elastic/elasticsearch/pull/66022#discussion_r553453278)).